routing/control_tower.go

```diff
@@ -110,6 +110,10 @@ type controlTower struct {
 	// that no race conditions occur in between updating the database and
 	// sending a notification.
 	paymentsMtx *multimutex.HashMutex
+
+	// newSubscriptionMutex makes sure no payment events are missed when
+	// a new subscription is made.
+	newSubscriptionMtx sync.RWMutex
```

Comment:
I think this commit should come before the release notes and changes to the RPC.

Reply:
Alright, I will fix that. I added that commit last, because Joost and I were discussing whether or not the locks were necessary. There's a different solution without additional locking, but it needs a comment, as shown here:
https://github.com/lightningnetwork/lnd/pull/6335#discussion_r847158161

If you could take a decision which strategy to follow, I can fix up the commits nicely.

Reply:
So yeah I removed the locks in the current version of this PR, I can reintroduce them if the concession with duplicates/out-of-order events is not what you'd want.